### PR TITLE
Remove Attachments from API

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #4555

## Description of Changes
- Removes all references of the Attachments table from the API

## Test Plan
- Unit tests are passing
- Should CA test affected functionality to check for regressions:
  - CRUD Comments
  - CRUD Discussion Drafts (*see Other Considerations)
  - CRUD Threads
  - Status route – confirm that on page init, this returns a success response
  - Pin thread
  - Lock thread
  - Update thread stage
  - viewComments route– confirm Thread View page renders comments successfully

## Deployment Plan
N/A

## Other Considerations
- Discussion Drafts appears to be a feature implemented in the API, but not in the UI. We use browser local storage for quill editor drafts. So there's nothing to test here.

### CODE REVIEW

File: packages/commonwealth/server/controllers/server_threads_methods/create_thread.ts [hunk 1]
```diff
@@ -18,16 +18,17 @@ import { ServerError } from 'near-api-js/lib/utils/rpc_errors';
...
 export const Errors = {
   InsufficientTokenBalance: 'Insufficient token balance',
   BalanceCheckFailed: 'Could not verify user token balance',
   ParseMentionsFailed: 'Failed to parse mentions',
-  NoBodyOrAttachments: 'Discussion posts must include body or attachment',
   LinkMissingTitleOrUrl: 'Links must include a title and URL',
   UnsupportedKind: 'Only discussion and link posts supported',
   FailedCreateThread: 'Failed to create thread',
   DiscussionMissingTitle: 'Discussion posts must include a title',
+  NoBody: '',
```
Comment (jnaviask) [inline packages/commonwealth/server/controllers/server_threads_methods/create_thread.ts:31]: Please add a description for the error.

File: packages/commonwealth/server/controllers/server_threads_methods/create_thread.ts [hunk 2]
```diff
@@ -55,43 +55,34 @@ export type CreateThreadResult = [
...
-  stage,
-  url,
-  attachments,
-  canvasAction,
-  canvasSession,
-  canvasHash,
-  discord_meta,
-}: CreateThreadOptions): Promise<CreateThreadResult> {
+export async function __createThread(
+  this: ServerThreadsController,
```
Comment (jnaviask) [inline packages/commonwealth/server/controllers/server_threads_methods/create_thread.ts:59]: Any reason we're now passing a `this` var?

**Review by jnaviask — CHANGES_REQUESTED** [review]
Fantastic work. Few small changes then g2g

File: packages/commonwealth/server/controllers/server_threads_methods/create_thread.ts (hunk 2 above)
Comment (rbennettcw) [reply packages/commonwealth/server/controllers/server_threads_methods/create_thread.ts:59]: It's for the TS intellisense in code editors so they know what the `this` value is for the function.

I forgot to add it previously so went ahead and did it in this PR.

Comment (rbennettcw) [reply packages/commonwealth/server/controllers/server_threads_methods/create_thread.ts:59]: It doesn't functionally change anything btw, it's strictly for typing purposes

**Review by jnaviask — APPROVED** [review]

**Comment by jnaviask** [comment]
cc @rbennettcw to resolve merge conflicts (not sure where they came from)

**Review by jnaviask — APPROVED** [review]

**Pull request merged**